### DISCUSSION TITLE
Old-school cloning over HTTPS of private repo

### DISCUSSION THREAD

**Discussion by hangarr** [community/community#23677] [New to GitHub]
<p>Hi.  I have not had to do this for some time, but the need has arisen.  I need to clone a (new) private repo using just HTTPS: .  When I do this, I get an error message that I’ve not seen before:</p>
<pre><code class="lang-auto">$ git clone https://hangarr09@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
Password for 'https://hangarr09@github.com': 
remote: Invalid username or password.
fatal: Authentication failed for 'https://hangarr09@github.com/hangarr/SoundPipeC.git/'
</code></pre>
<p>This is on a Mac running El Capitan with git from Xcode for El Capitan.  Just for completeness, I also seem to get anomalous results when I do this (note it doesn’t as for credentials):</p>
<pre><code class="lang-auto">$ git clone https://github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
remote: Repository not found.
fatal: repository 'https://github.com/hangarr/SoundPipeC.git/' not found

$ git clone https://hangarr@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
remote: Repository not found.
fatal: repository 'https://hangarr@github.com/hangarr/SoundPipeC.git/' not found
</code></pre>
<p>and:</p>
<pre><code class="lang-auto">$ git clone https://hangarr09:xxxxxxxx@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
remote: Invalid username or password.
fatal: Authentication failed for 'https://hangarr09:xxxxxxxx@github.com/hangarr/SoundPipeC.git/'
</code></pre>
<p>(Obviously I’ve masked the password.)</p>
<p>Any ideas what’s going wrong here?  I can’t use SSH on this machine for non-technical reasons and I don’t have 2FA enabled as far as I can see.  Thanks very much for any help.</p>

**Comment by airtower-luna** [comment]
<p>The documentation on <a href="https://docs.github.com/en/free-pro-team@latest/github/using-git/which-remote-url-should-i-use#cloning-with-https-urls">Cloning with HTTPS URLs</a> has been stating this for a while:</p>
<blockquote>
<p>Password-based authentication for Git is deprecated, and we recommend using a personal access token (PAT) when prompted for a password instead, which is more secure. Treat your token just like a password. For more information, see “<a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">Creating a personal access token</a>.”</p>
</blockquote>
<p>Does it work if you use a PAT instead of your password?</p>

**Comment by hangarr** [comment]
<p>Thanks.</p>
<p>Unfortunately, that seems to have uncovered a second issue.  Note the error message here:</p>
<pre><code class="lang-auto">$ git clone https://hangarr09@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
Password for 'https://hangarr09@github.com':   *(PAT entered here)*
remote: Repository not found.
fatal: repository 'https://hangarr09@github.com/hangarr/SoundPipeC.git/' not found
</code></pre>
<p>Here’s the message with an invalid PAT:</p>
<pre><code class="lang-auto">$ git clone https://hangarr09@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
Password for 'https://hangarr09@github.com': 
remote: Invalid username or password.
fatal: Authentication failed for 'https://hangarr09@github.com/hangarr/SoundPipeC.git/'
</code></pre>
<p>And just for completeness again:</p>
<pre><code class="lang-auto">Wellspri2:Project 1 hangartner$ git clone https://hangarr09:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
remote: Repository not found.
fatal: repository 'https://hangarr09:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx@github.com/hangarr/SoundPipeC.git/' not found

Wellspri2:Project 1 hangartner$ git clone https://github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
remote: Repository not found.
fatal: repository 'https://github.com/hangarr/SoundPipeC.git/' not found
</code></pre>
<p>Is GitHub possibly allowing me to create a private repo but not clone it?  This is a personal (free) account being used here to share an example of some old work with another single entity.  To my recollection, in the old days GitHub didn’t allow one to create a private repo on a free account.</p>
<p>Thanks for any insights.</p>

**Comment by airtower-luna** [comment]
<p>Hm, the only reason I can think of would be that the PAT lacks the <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes"><code>repo</code> scope</a>. Without that you can only read public data. You can check and if necessary update that on the same settings page where you can generate new tokens.</p>
<aside class="quote no-group" data-username="hangarr" data-post="3" data-topic="140430">
<div class="title">
<div class="quote-controls"></div>
 hangarr:</div>
<blockquote>
<p>To my recollection, in the old days GitHub didn’t allow one to create a private repo on a free account.</p>
</blockquote>
</aside>
<p>I remember that was the case, but that’s changed, see <a href="https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/githubs-products#github-free-for-user-accounts">GitHub Free for user accounts</a>.</p>

**Comment by hangarr** [comment]
<p>Thanks for your assistance.  What you’ve suggested makes perfect sense but it appears there may be some other problem with GitHub and private repos.  Although I’m sure I gave the first PAT repo scope, just to confirm I created a second with (only) repo scope and got the same failure:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/6/465755425688608da78fd24b2ac499f174315794.png" data-download-href="/uploads/short-url/a2gx3i20mxAvc1FlEm2sCbW0fYg.png?dl=1" title="githubpat" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2850996/181078718-9fbfc24a-90a6-4f69-8701-4eea766260fc.png" alt="githubpat" data-base62-sha1="a2gx3i20mxAvc1FlEm2sCbW0fYg" width="690" height="386" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/6/465755425688608da78fd24b2ac499f174315794_2_690x386.png, https://aws1.discourse-cdn.com/github/original/3X/4/6/465755425688608da78fd24b2ac499f174315794.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/6/465755425688608da78fd24b2ac499f174315794.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/6/465755425688608da78fd24b2ac499f174315794_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">githubpat</span><span class="informations">1034×579 48.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Perhaps this needs to be brought to the attention of some team at GitHub?  Any ideas how to do that?  Thanks again for your assistance.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="hangarr" data-post="3" data-topic="140430">
<div class="title">
<div class="quote-controls"></div>
 hangarr:</div>
<blockquote>
<pre><code class="lang-auto">remote: Repository not found.
fatal: repository 'https://hangarr09@github.com/hangarr/SoundPipeC.git/' not found
</code></pre>
</blockquote>
</aside>
<p>I just noticed one more thing: The user account you’re using for authentication and the one owning the repository aren’t the same. Does the former actually have access to the private repository?</p>
<aside class="quote no-group" data-username="hangarr" data-post="5" data-topic="140430">
<div class="title">
<div class="quote-controls"></div>
 hangarr:</div>
<blockquote>
<p>Perhaps this needs to be brought to the attention of some team at GitHub? Any ideas how to do that? Thanks again for your assistance.</p>
</blockquote>
</aside>
<p>There are some Github people on this forum, I would hope one of them sees this. I’ve moved this thread to the <a class="hashtag" href="/c/how-to-use-git-and-github/21">#<span>how-to-use-git-and-github</span></a> section to hopefully make that easier. :wink:</p>
<p>Or you could try the contact form: <a href="https://support.github.com/request">https://support.github.com/request</a></p>

**Comment by hangarr** [comment]
<p>Thanks very much for the additional thoughts.</p>
<p>You caught something that also has been an unanswered question for me.  This repo was created sometime ago with the user name “hangarr” that corresponds to an email account “<a href="mailto:hangarr@yyyyy.com">hangarr@yyyyy.com</a>”  on a domain <a>yyyyy.com</a> I own.  I made that email account the secondary email account on this GitHub account.   I added a public account hangarr09 account as the primary email account.  I can log in with both email addresses for some reason.  That has turned out to be convenient if puzzling.</p>
<p>Just to be safe, I tried adding both accounts as collaborators on the repo and this went through without a hiccup by Github but without any notification to either email account.  I can’t seem to find a list of collaborators on this repo but I probably don’t know where to look these days.</p>
<p>I get the same error message trying both with one difference:</p>
<pre><code class="lang-auto">$ git clone https://hangarr09@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
Password for 'https://hangarr09@github.com': 
remote: Repository not found.
fatal: repository 'https://hangarr09@github.com/hangarr/SoundPipeC.git/' not found

$ git clone https://hangarr@github.com/hangarr/SoundPipeC.git
Cloning into 'SoundPipeC'...
remote: Repository not found.
fatal: repository 'https://hangarr@github.com/hangarr/SoundPipeC.git/' not found
</code></pre>
<p>Notice the first asks for a password and the second gives up (immediately).  I’m assuming the first at least is getting through authentication and the second is saying that the actual repo “username” is not a valid for cloning the private repo.</p>
<p>Thanks also for explaining how GitHub folks monitor the communities and for moving this to the correct section.  I appreciate your efforts on this.</p>

**Comment by byrneh** [comment, marked as answer]
<p>I don’t think “hangarr09” is an actual GitHub usename as it is not found when searched for on GitHub.<br>
A user “hangarr” is found.<br>
Your write up is a little confusing, but perhaps you have two emails registered on single github account username “hangarr09”.<br>
So firstly the clone must use an actual  GitHub username.</p>
<p>Cloning a private repository using a correctly scoped PAT definitely works when the user have read access to the repository if the repository is owned by th user doing the clone you definetly have pwrmiasions. Easily checked by using the the gui and and accessing the repository, you can also copy the clone URL format<br>
So setting up a correctly scoped Pat on this user and uisbg it for the clone should work.</p>

**Comment by hangarr** [comment]
<p>Thanks very much for your response and the additional info.</p>
<p>You’re correct there is a primary and secondary email associated with this Github account.   I can login on the website with the username or  either email account name.  While that is a good thing, it turns out it confused me.</p>
<p>Your clarification that I can only clone with the username was the key.  When I used the username with a PAT it worked.</p>
<p>Thanks both of you for your help.</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/hangarr">@hangarr</a>, glad you are working now,  you can mark the answer as a solution if this is now solved</p>
